daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -517,11 +517,8 @@ private[speedy] case class PartialTransaction(
       byInterface,
       version,
     )
-    mustBeActive(
```

Comment:
afaict you are not actually adding a replacement for this. I assume that’s not necessary because we check this before calling this already. Do you think it would make sense to keep `mustBeActive` anyway (in both cases here) as an assertion that throws an internal error as a fail-safe?

Reply:
I am not very fan of checking things twice.
I give you a wrong sens of security, make the code more complicate than necessary, and also more difficult to understand (why this property is check twice and this once not).  

Not feeling strongly about that, thought. I let you  judge for it. 
Should I let `mustActive` (and throw an internal error) or should I drop it completely ?

Reply:
Checking twice is definitely a bad idea if you treat both of them as actual validations because then you’re just doing shotgun validation which never works. If you make a clear distinction between "this is the actual check" and "this is an assertion which checks something that should always hold" I think it can make sense and help at least during refactoring when you violate that assumption (e.g. imagine for whatever reason exercise no longer ends up going through SBUFetch afterwards).

So I’d add it but make it an internal error/assertion failure but your choice.